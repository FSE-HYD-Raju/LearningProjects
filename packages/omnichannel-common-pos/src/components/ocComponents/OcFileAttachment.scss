.OcFileAttachment {
	&-container {
		@include flex-container(row, stretch);
		width: 100%;
		align-items: initial;
		padding: 0 0 0.5rem 0;
	}
	&-label {
		min-width: 10.25rem;
		color: $gray-600;
		margin: 0;
	}
	&-data {
		display: flex;
		flex-direction: column;
		max-width: 24.65rem;
		width: 100%;
	}
	&-file-form {
		display: flex;
		align-items: center;
		padding: 0.5rem 0;
		flex-wrap: wrap;
		input.OcFileAttachment-hidden-input {
			max-width: 5rem;
			opacity: 0;
		}
	}
	&-attach-file-btn {
		margin-left: -5rem;
		padding: 0;
		font-size: $font-size-sm;
	}
	&-file-uploaded {
		border-bottom: 0.0625rem solid $weasel-border-color-solid;
		display: flex;
		padding: 0.5rem 0;
		button {
			padding: 0 0.565rem;
		}
	}
	&-file-name {
		max-width: 21rem;
		white-space: pre-wrap;
		word-wrap: break-word;
		word-break: break-all;
	}
	&-attachment-hint {
		color: $gray-600;
		font-size: $font-size-sm;
		white-space: pre-wrap;
		word-wrap: break-word;
		word-break: break-word;
		width: 100%;
		padding: 0.25rem 0;
	}
	@include media-breakpoint-up(sm) {
		&-label {
			padding: 0.5rem 0;
		}
	}
}
