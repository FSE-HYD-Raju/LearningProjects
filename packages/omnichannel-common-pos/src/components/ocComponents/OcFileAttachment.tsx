import * as R from "react";
import FormattedMessage from "../../channelUtils/FormattedMessage";
import cssns from "../../utils/cssnsConfig";
import { contextTypesValidationMap, ContextType } from "../../types/ContextType";
import messages from "./ocFileAttachment.messages";
import { OcButton, OcButtonType } from "./button/OcButton";
const React = cssns("OcFileAttachment").React as typeof R;

interface OcFileAttachmentProps {
	maxFiles?: number;
}
interface OcFileAttachmentState {
	files: Array<File>;
	disabled: boolean;
}

class OcFileAttachment extends (React as typeof R).Component<OcFileAttachmentProps, OcFileAttachmentState> {
	static contextTypes = contextTypesValidationMap;

	static defaultProps = {
		maxFiles: 1,
	};

	constructor(props: OcFileAttachmentProps, context: ContextType) {
		super(props, context);
		this.state = {
			files: [],
			disabled: false,
		};
	}

	validateFileSize = (files: Array<File>) => {
		return files.filter((file: File) => {
			return file.size <= 20971520;
		});
	};
	getFiles(): File[] {
		return this.state.files;
	}

	handleFileUpload = (event: any) => {
		const uploadedFiles: Array<File> = Array.from(event.target.files);
		const maxFiles = this.props.maxFiles!;
		event.target.value = "";
		this.setState(({ files }) => {
			const newFiles = files.length >= maxFiles ? files : files.concat(uploadedFiles);
			const validatedFiles = this.validateFileSize(newFiles);
			return {
				files: validatedFiles,
				disabled: validatedFiles.length >= maxFiles,
			};
		});
	};

	handleFileDelete = (passedIndex: number) => {
		this.setState(({ files }) => {
			const updatedFiles = files.filter((file, index) => index !== passedIndex);
			const maxFiles = this.props.maxFiles!;
			return {
				files: updatedFiles,
				disabled: updatedFiles.length >= maxFiles,
			};
		});
	};

	renderFileList = (files: Array<File>) => {
		return files.map((file: File, index: number) => {
			return (
				<div key={index} className="file-uploaded">
					<span className="file-name">{file.name}</span>
					<OcButton
						buttonType={OcButtonType.LINK}
						id="OcFileAttachment-delete-btn"
						htmlBtnType="button"
						onClick={() => this.handleFileDelete(index)}
					>
						<i className="fas fa-times" />
					</OcButton>
				</div>
			);
		});
	};

	renderAttachmentHint = (maxFiles: OcFileAttachmentProps["maxFiles"]) => {
		return this.context.intl.formatMessage(messages.fileAttachmentHintMessage, { maxFiles });
	};

	render() {
		return (
			<div className="container">
				<div className="label">
					<FormattedMessage {...messages.fileAttachmentInputLabel} />
				</div>
				<div className="data">
					<div className="file-list" id="OcFileAttachment-files-uploaded-list">
						{this.renderFileList(this.state.files)}
					</div>
					<form className="file-form">
						<input
							disabled={this.state.disabled}
							className="OcFileAttachment-hidden-input"
							id="OcFileAttachment-fileItem"
							type="file"
							onChange={this.handleFileUpload}
						/>
						<OcButton
							buttonType={OcButtonType.LINK}
							className="attach-file-btn"
							id="OcFileAttachment-fileItem-btn"
							htmlBtnType="button"
							disabled={this.state.disabled}
						>
							<FormattedMessage {...messages.fileAttachmentAttachFileBtnLabel} />
						</OcButton>
						<div className="attachment-hint">{this.renderAttachmentHint(this.props.maxFiles)}</div>
					</form>
				</div>
			</div>
		);
	}
}

export default OcFileAttachment;
export { OcFileAttachmentProps, OcFileAttachmentState };
