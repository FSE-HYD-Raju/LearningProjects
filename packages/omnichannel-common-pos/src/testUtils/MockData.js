import _ from "lodash";
import R from "ramda";

const firstDateThisWeek = () => {
	const date = new Date();
	while (date.getDay() !== 0) {
		date.setDate(date.getDate() - 1);
	}
	return date;
};

const people = [
	{
		firstName: "Kathrin",
		lastName: "K",
		gender: "female",
		language: "swe",
		countryOfBirth: "SE",
		birthDay: firstDateThisWeek(),
		emails: [{ email: "kathrin.k@qvantel.com" }],
		email: "kathrin.k@qvantel.com",
		mobileNumbers: [{ number: "2398457" }],
		mobileNumber: "2398457",
		fixedLineNumbers: [{ number: "2398457" }],
		fixedLineNumber: "2398457",
		password: "qwerasdf",
		passwordConfirmation: "qwerasdf"
	},
	{
		firstName: "John",
		lastName: "Smith",
		gender: "male",
		language: "fin",
		birthDay: firstDateThisWeek(),
		countryOfBirth: "United Kingdom",
		emails: [{ email: "john.smith@qvantel.com" }],
		email: "john.smith@qvantel.com",
		mobileNumbers: [{ number: "3498573" }],
		mobileNumber: "3498573",
		fixedLineNumbers: [{ number: "52497584" }],
		fixedLineNumber: "52497584"
	}
];

const basketItems = [
	{
		id: "7879c247-9e98-40ee-bb48-544eb366494a",
		type: "basketItems",
		attributes: {
			product: {
				name: "Movie Steaming Service"
			},
			totalPrices: [
				{
					type: "ONE_TIME",
					taxFreeAmount: 1.9,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: 14.9,
					taxRate: 0,
					currency: "EUR"
				}
			],
			childBasketItems: [],
			totalUpfrontPrice: 1.9,
			totalUpfrontPriceWithChildren: 1.9
		}
	},
	{
		id: "c0065ffc-77f4-4e44-b15c-fbd2f78d04ce",
		type: "basketItems",
		attributes: {
			product: {
				name: "XDSL"
			},
			totalPrices: [
				{
					type: "ONE_TIME",
					taxFreeAmount: 100.6,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: 30,
					taxRate: 0,
					currency: "EUR"
				}
			],
			childBasketItems: []
		}
	}
];

const deliveryItem = {
	id: "c0065ffc-77f4-4e44-b15c-fbd2f78d04ce",
	type: "basketItems",
	inputtedCharacteristics: {
		CH_Delivery_Contact_Info: "5c5dddf6-72be-4d0a-9711-6027f247af28"
	},
	product: {
		name: "Home delivery",
		id: "PO_Home_Delivery"
	}
};

export default class MockData {
	static getPerson(searchCriteria) {
		if (searchCriteria) {
			return _.find(people, searchCriteria);
		} else {
			return people[0];
		}
	}

	static getPeople() {
		return people;
	}

	static getPostalAddress() {
		return {
			street: "1 Test St",
			coAddress: "Best Ltd.",
			postalCode: "90210",
			city: "Test City",
			country: "FI"
		};
	}

	static getUser() {
		return {
			"id": "bf588945-ce95-490a-a774-8ef91661a146",
			"type": "persons",
			"attributes": {
				"lastName": "Testeri",
				"addresses": null,
				"gender": "male",
				"fixedLineNumbers": [],
				"formattedName": "Teppo Testeri",
				"roles": null,
				"billingAccountId": null,
				"language": null,
				"emails": [
					{
						"role": "PRIMARY",
						"id": "2ab79d7b-862d-49df-8afc-e6104a914919",
						"email": "tepsteri@tepinposti.fi"
					}
				],
				"honorificSuffix": null,
				"countryOfBirth": "FI",
				"personRelationships": [],
				"birthDay": null,
				"placeOfBirth": null,
				"characteristics": null,
				"mobileNumbers": [
					{
						"role": "PRIMARY",
						"id": "a843f46d-f729-40dd-a7ce-240a64bc008e",
						"number": "1234576890"
					}
				],
				"honorificPrefix": null,
				"avatar": "/img/icon-user-male.svg",
				"privacySettings": {
					"privacy1": false,
					"privacy2": false
				},
				"postalAddresses": [
					{
						"role": "PRIMARY",
						"id": "5e30221d-ad19-4043-8088-f594f5fdf77f",
						"street": "Piippukatu 11 A 1",
						"coAddress": null,
						"postalCode": "40100",
						"city": "Roma",
						"country": "FI",
						"county": null,
						"description": null,
						"postOfficeBox": null,
						"stateOrProvince": null,
						"addressRegisterId": null,
						"apartment": null,
						"building": null
					},
					{
						"role": "DELIVERY",
						"id": "5c5dddf6-72be-4d0a-9711-6027f247af28",
						"street": "Piippukatu 11 A 1",
						"coAddress": "Piippukatu 11 A 1",
						"postalCode": "40100",
						"city": "Roma",
						"country": "FI",
						"county": null,
						"description": null,
						"postOfficeBox": null,
						"stateOrProvince": "",
						"addressRegisterId": null,
						"apartment": null,
						"building": null
					}
				],
				"identifications": [],
				"firstName": "Teppo",
				"customerAccountId": null,
				"nationality": null,
				"additionalName": null,
				"maritalStatus": null
			}
		}
	}

	static getBasket(withItems: boolean, withDeliveryItems: boolean) {
		const basket = {
			activeBasket: {
				attributes: {
					totalPrices: [],
					totalUpfrontPrice: 0
				}
			}
		};

		if (withItems) {
			basket.basketItems = basketItems;

			const totalPriceOneTime = R.reduce(
				R.add,
				0,
				basketItems.map(bi => {
					return bi.attributes.totalPrices[0].taxFreeAmount;
				})
			);
			const totalPriceMonthly = R.reduce(
				R.add,
				0,
				basketItems.map(bi => {
					return bi.attributes.totalPrices[0].taxFreeAmount;
				})
			);

			basket.activeBasket.attributes.totalPrices = [
				{
					type: "ONE_TIME",
					taxFreeAmount: totalPriceOneTime,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: totalPriceMonthly,
					taxRate: 0,
					currency: "EUR"
				}
			];

			if (withDeliveryItems) {
				basket.basketItems[0].attributes.childBasketItems.push(
					deliveryItem
				);
			}
		}

		return basket;
	}
}
