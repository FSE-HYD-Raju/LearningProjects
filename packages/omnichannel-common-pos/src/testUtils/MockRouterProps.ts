import {
	RouteComponentProps,

} from "react-router";

import {
	Href,
	UnregisterCallback,
	TransitionPromptHook,
	LocationState,
	Path,
	LocationDescriptorObject,
	LocationListener
} from "history";

const mockLocation: RouteComponentProps<any>["location"] = {
		hash: "",
		key: "",
		pathname: "",
		search: "",
		state: {}
};

const mockHistory: RouteComponentProps<any>["history"] = {
	length: 2,
	action: "POP",
	location: mockLocation,
	push: (path: Path | LocationDescriptorObject, state?: LocationState): void => {},
	replace: (path: Path | LocationDescriptorObject, state?: LocationState): void => {},
	go: (n: number): void => {},
	goBack: (): void => {},
	goForward: (): void => {},
	block: (prompt?: boolean | string | TransitionPromptHook): UnregisterCallback => { return () => {}; },
	listen: (listener: LocationListener): UnregisterCallback => {return () => {}; },
	createHref: (location: LocationDescriptorObject): Href => { return ""; },
};

const mockRouterProps: RouteComponentProps<any> = {
		match: {
			isExact: true,
			params: {},
			path: "",
			url: ""
		},
		location: mockLocation,
		history: mockHistory,
		staticContext: {}
};

export {
	mockRouterProps,
	mockLocation,
	mockHistory,
	RouteComponentProps
};
