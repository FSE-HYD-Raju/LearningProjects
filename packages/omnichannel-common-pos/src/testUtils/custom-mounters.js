import React from "react";
import { IntlProvider, intlShape } from "react-intl";
import { mount, shallow } from "enzyme";
import PropTypes from "prop-types";
import R from "ramda";
import TestUtils from "./TestUtils";

const MAX_DIVE_DEEP_LEVEL = 16;
const intlProvider = new IntlProvider({ locale: "en" }, {});
const { intl } = intlProvider.getChildContext();

const baseContext = {
	flux: {
		persist: () => {}
	},
	store: TestUtils.mockReduxStore(),
	intl,
	location: {},
	only: {},
	skip: {}
};

const nodeWithIntlProp = (node, mockedRouterOptions) => {
	if (mockedRouterOptions) {
		mockedRouterOptions.props().history.push = jest.fn();
	}
	return React.cloneElement(node, {
		intl,
		...(mockedRouterOptions ? mockedRouterOptions.props() : {})
	});
};

function getOptionsForShallowMount(options, debug, mockedRouterContextOptions) {
	const mountOptions = R.merge(options ? options : {}, {
		disableLifecycleMethods: !(
			options && options.disableLifecycleMethods === false
		),
		context: R.mergeAll([
			debug
				? {
						...baseContext
					}
				: baseContext,
			R.omit("intl", options && options.context ? options.context : {}),
			mockedRouterContextOptions.context
		]),
		childContextTypes: R.merge(
			{
				intl: intlShape,
				store: PropTypes.object
			},
			mockedRouterContextOptions.childContextTypes
		)
	});
	TestUtils.mockReduxStoreToFlux(mountOptions.context);
	return mountOptions;
}
function deepDive(wrapper) {
	let diveAttemptsCount = 0;
	while (
		wrapper.hostNodes().length === 0 &&
		diveAttemptsCount < MAX_DIVE_DEEP_LEVEL
	) {
		wrapper = wrapper.dive();
		diveAttemptsCount += 1;
	}
	return wrapper;
}
export function shallowWithContextAndRouterProps(node, options, debug) {
	return shallowWithContext(node, options, debug, true);
}
export function shallowWithContextAndLifecycle(node, options, debug) {
	return shallowWithContext(
		node,
		R.merge(options || {}, { disableLifecycleMethods: false }),
		debug,
		true
	);
}
export function deepShallowWithContextAndLifecycle(node, options, debug) {
	return deepDive(shallowWithContextAndLifecycle(node, options, debug));
}
export function deepShallowWithContext(node, options, debug) {
	return deepDive(shallowWithContext(node, options, debug, true));
}
export function shallowWithContext(node, options, debug, injectRouterProps) {
	const mockedRouterOptions = TestUtils.getMockedRouterOptions();
	const mockedRouterContextOptions = mockedRouterOptions.get();
	return shallow(
		nodeWithIntlProp(node, injectRouterProps ? mockedRouterOptions : null),
		getOptionsForShallowMount(options, debug, mockedRouterContextOptions)
	);
}

export function mountWithContextAndRouterProps(node, options, debug) {
	return mountWithContext(node, options, debug, true);
}
export function mountWithContext(node, options, debug, injectRouterProps) {
	const mockedRouterOptions = TestUtils.getMockedRouterOptions();
	const mockedRouterContextOptions = mockedRouterOptions.get();
	const mountOptions = R.merge(options ? options : {}, {
		context: R.mergeAll([
			debug
				? {
						...baseContext
					}
				: baseContext,
			R.omit("intl", options && options.context ? options.context : {}),
			mockedRouterContextOptions.context
		]),
		childContextTypes: R.merge(
			{
				flux: PropTypes.object,
				store: PropTypes.object,
				intl: intlShape,
				location: PropTypes.object,
				only: PropTypes.object,
				skip: PropTypes.object
			},
			mockedRouterContextOptions.childContextTypes
		)
	});
	TestUtils.mockReduxStoreToFlux(mountOptions.context);
	return mount(
		nodeWithIntlProp(node, injectRouterProps ? mockedRouterOptions : null),
		mountOptions
	);
}

const attachmentElementId = "__OC_COMPONENT_TEST_ATTACHMENT_ELEMENT__";

export function attachWithContext(node, options) {
	const attachmentElement = document.createElement("div");
	attachmentElement.setAttribute("id", attachmentElementId);
	document.body.appendChild(attachmentElement);

	const wrapper = mountWithContext(
		node,
		R.merge(options, { attachTo: attachmentElement })
	);
	const doppelWrapper = hijackDetacher(wrapper);

	return doppelWrapper;
}

function hijackDetacher(wrapper) {
	const doppel = wrapper;
	doppel.__detach = wrapper.detach;

	doppel.detach = () => {
		removeAttachmentElements();
		doppel.__detach.apply(wrapper);
	};

	return doppel;
}

function removeAttachmentElements() {
	document.body.querySelectorAll("#" + attachmentElementId).forEach(elem => {
		document.body.removeChild(elem);
	});
}

export function detach(wrapper) {
	removeAttachmentElements();
	wrapper.detach();
}
