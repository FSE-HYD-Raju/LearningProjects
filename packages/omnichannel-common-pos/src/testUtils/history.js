import isClient from "../utils/isClient";
import { createBrowserHistory, createMemoryHistory } from "history";
import { useRouterHistory } from "react-router";

const history = isClient
	? useRouterHistory(createBrowserHistory)({ queryKey: false })
	: createMemoryHistory();

export default history;
