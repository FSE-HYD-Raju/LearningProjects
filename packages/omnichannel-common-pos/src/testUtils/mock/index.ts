import { default as productOffering } from "./ProductOfferingMock";
import { ProductMock as product } from "./ProductMock";
import { AgreementMock as agreement } from "./AgreementMock";
import { ChargingBalancesMock as chargingBalances } from "./ChargingBalancesMock";
import { default as postalAddress } from "./PostalAddressMock";
import { default as identification } from "./IdentificationMock";
import { default as user } from "./UserMock";
import { default as order } from "./OrderMock";
import { default as basket } from "./BasketMock";
export { productOffering, postalAddress, identification, user, order, basket, product, agreement, chargingBalances };
