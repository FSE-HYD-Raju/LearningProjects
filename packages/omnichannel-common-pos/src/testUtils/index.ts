import ReactWidgetsTestUtils from "./ReactWidgetsTestUtils";
import * as MockDataMaker from "./mock";
const {
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach
} = require("./custom-mounters");
const TestUtils = require("./TestUtils");
const MockData = require("./MockData");
const SimpleDataMock = require("./mock/simpleDataMock");

export {
	MockData,
	ReactWidgetsTestUtils,
	TestUtils,
	SimpleDataMock,
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach,
	MockDataMaker
};

export * from "./MockRouterProps";

export default TestUtils;
