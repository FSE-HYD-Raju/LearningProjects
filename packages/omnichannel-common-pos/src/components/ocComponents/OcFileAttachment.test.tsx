import * as React from "react";
import { mountWithContext, shallowWithContext } from "../../testUtils";
import OcFileAttachment, { OcFileAttachmentProps } from "./OcFileAttachment";

describe("OcFileAttachment", () => {
	let props: OcFileAttachmentProps;
	beforeEach(()  => {
		props = {
			maxFiles: 3
		};
	});

	it("succeeds at shallow mount with minimum props", () => {
		const wrapper = shallowWithContext(
			<OcFileAttachment {...props} />
		);
		expect(wrapper).toMatchSnapshot();
	});

	it("succeeds at deep mount with minimum props", () => {
		mountWithContext(<OcFileAttachment {...props} />);
	});

	it("disables button when max files size is exceeded", () => {
		const wrapper = mountWithContext(<OcFileAttachment {...props} />);
		wrapper.setState({
			disabled: true
		});
		wrapper.update();
		const attachBtn = wrapper.find("#OcFileAttachment-fileItem-btn").at(0);
		expect(attachBtn.props().disabled).toBe(true);

	});

	it("removes file whe ndelete button is pressed", () => {
		const wrapper = mountWithContext(<OcFileAttachment {...props} />);
		wrapper.setState({
			files: [{name: "someName", size: 1024}]
		});
		wrapper.update();
		const deleteBtn = wrapper.find("#OcFileAttachment-delete-btn").at(0);
		deleteBtn.simulate("click");
		wrapper.update();
		const fileList = wrapper.find("#OcFileAttachment-files-uploaded-list").at(0);
		expect(fileList.props().children).toHaveLength(0);
	});
});
