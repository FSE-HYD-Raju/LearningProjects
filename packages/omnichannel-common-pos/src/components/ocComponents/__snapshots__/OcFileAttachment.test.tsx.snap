// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OcFileAttachment succeeds at shallow mount with minimum props 1`] = `
<div
  className="OcFileAttachment-container"
>
  <div
    className="OcFileAttachment-label"
  >
    <FormattedMessageWrapper
      defaultMessage="Attachment"
      description="File attachment, label for file attachment input"
      id="oc-file-attachment-input-label"
      wrapperType="wrapper"
    />
  </div>
  <div
    className="OcFileAttachment-data"
  >
    <div
      className="OcFileAttachment-file-list"
      id="OcFileAttachment-files-uploaded-list"
    />
    <form
      className="OcFileAttachment-file-form"
    >
      <input
        className="OcFileAttachment-hidden-input"
        disabled={false}
        id="OcFileAttachment-fileItem"
        onChange={[Function]}
        type="file"
      />
      <OcButton
        buttonType="LINK"
        className="OcFileAttachment-attach-file-btn"
        disabled={false}
        htmlBtnType="button"
        id="OcFileAttachment-fileItem-btn"
      >
        <FormattedMessageWrapper
          defaultMessage="Attach file"
          description="File attachment, label for attach button"
          id="oc-file-attachment-attach-file-btn-label"
          wrapperType="wrapper"
        />
      </OcButton>
      <div
        className="OcFileAttachment-attachment-hint"
      >
        3 attachments max. File size shouldn't exceed 20Mb.
      </div>
    </form>
  </div>
</div>
`;
